### PR TITLE
Support BatchNorm3D layer on IE backend

### PR DESCRIPTION
**Pull request by l-bat**
**Merge with extra:** opencv/opencv_extra#602

<cut/>

```
allow_multiple_commits=1

force_builders=Linux AVX2,Custom,Mac OpenCL

buildworker:Mac OpenCL=macosx-1
build_image:Mac OpenCL=openvino-2019r1
test_modules=dnn

buildworker:Custom=linux-2
docker_image:Custom=ubuntu-openvino-2019r1:16.04
test_opencl:Custom=ON
test_bigdata:Custom=1
test_filter:Custom=*
test_modules:Custom=dnn

ci_branch:Linux AVX2=no-checks
docker_image:Linux AVX2=ubuntu-openvino-2019r1:16.04
test_opencl:Linux AVX2=OFF
buildworker:Linux AVX2=linux-1
build_parallel_tests:Linux AVX2=1
test_bigdata:Linux AVX2=1
test_module:Linux AVX2=dnn,python2,python3,java
tests_filter:Linux AVX2=*
```

### CODE REVIEW

File: modules/dnn/src/layers/batch_norm_layer.cpp [hunk 1]
```diff
@@ -150,9 +153,9 @@ class BatchNormLayerImpl CV_FINAL : public BatchNormLayer
 
     virtual bool supportBackend(int backendId) CV_OVERRIDE
     {
-        return backendId == DNN_BACKEND_OPENCV ||
+        return (backendId == DNN_BACKEND_OPENCV && (dims == 4 || dims == 2)) ||
```
Comment (dkurt) [inline modules/dnn/src/layers/batch_norm_layer.cpp]: Have you tried to run BN with 5D input? Looks a bit confusing for me why it works with 4 and 2 dimensions but does not work with 3 and 5.

File: modules/dnn/src/layers/batch_norm_layer.cpp [hunk 2]
```diff
@@ -178,11 +181,11 @@ class BatchNormLayerImpl CV_FINAL : public BatchNormLayer
...
 
         UMat &inpBlob = inputs[0];
-        CV_Assert(inpBlob.dims == 2 || inpBlob.dims == 4);
         int groups = inpBlob.size[0];
         int channels = inpBlob.size[1];
-        int rows = inpBlob.dims > 2 ? inpBlob.size[2] : 1;
-        int cols = inpBlob.dims > 2 ? inpBlob.size[3] : 1;
+        int depth = inpBlob.dims == 5 ? inpBlob.size[2] : 1;
+        int rows = inpBlob.dims > 2 ? inpBlob.size[inpBlob.dims - 2] : 1;
+        int cols = inpBlob.dims > 2 ? inpBlob.size[inpBlob.dims - 1] : 1;
```
Comment (dkurt) [inline modules/dnn/src/layers/batch_norm_layer.cpp]: Why not just compute a total number of elements in spatial dimensions?

```cpp
int planeSize = total(inpBlob.size, 2);
Mat inpBlobPlane(1, planeSize, CV_32F, inpBlob.ptr<float>(num, n));
Mat outBlobPlane(1, planeSize, CV_32F, outBlob.ptr<float>(num, n));
...

```

Comment (l-bat) [reply modules/dnn/src/layers/batch_norm_layer.cpp]: done
